step_implementations/GoFunctionalTests/context/materials.rb

```diff
@@ -0,0 +1,60 @@
+##########################################################################
+# Copyright 2016 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+require "open3"
+
+module GoFunctionalTests
+  module Context
+    class Materials
+      include FileUtils
+      def initialize (*args)
+      end
+    end
+
+    class GitMaterials < Materials
+
+      def initialize(*args)
+        super()
+        @path = "#{Dir.tmpdir()}/gitmaterial"
+      end
+
+      def setup_material_for(pipeline)
+
+        rm_rf(@path) if Dir.exists?(@path)
```

Comment:
`rm_rf` implies no check for existance is needed, `rm` will fail if the directory does not exist. So this code feels redundant.